services/processor/native/repository/_Elections/processing_rewards.go

```diff
@@ -145,38 +144,48 @@ func getExcellenceProgramGuardians() []byte {
 	return state.ReadBytes(_formatExcellenceProgramGuardians())
 }
 
-func _setExcellenceProgramGuardians(guardians map[[20]byte]uint64) {
+func _setExcellenceProgramGuardians(guardians guardianArray) {
 	guardiansForSave := make([]byte, 0, len(guardians)*20)
-	for guardianAddr := range guardians {
-		guardiansForSave = append(guardiansForSave, guardianAddr[:]...)
+	for _, guardian := range guardians {
+		guardiansForSave = append(guardiansForSave, guardian.address[:]...)
 	}
 	state.WriteBytes(_formatExcellenceProgramGuardians(), guardiansForSave)
 }
 
 /***
  * Rewards: Sort top guardians using sort.Interface
  */
-func _getTopGuardians(guardiansAccumulatedStake map[[20]byte]uint64) (topGuardiansStake map[[20]byte]uint64, totalVotes uint64) {
+func _getTopGuardians(guardiansAccumulatedStake map[[20]byte]uint64) (topGuardiansStake guardianArray, totalVotes uint64) {
 	totalVotes = uint64(0)
-	topGuardiansStake = make(map[[20]byte]uint64)
 
 	guardianList := make(guardianArray, 0, len(guardiansAccumulatedStake))
 	for guardian, vote := range guardiansAccumulatedStake {
 		guardianList = append(guardianList, &guardianVote{guardian, vote})
 	}
 	sort.Sort(guardianList)
 
-	for i := 0; i < ELECTION_GUARDIAN_EXCELLENCE_MAX_NUMBER; i++ {
-		fmt.Printf("elections %10d rewards: top guardian %x, has %d votes\n", _getCurrentElectionBlockNumber(), guardianList[i].guardian, guardianList[i].vote)
+	i := 0
+	for i = 0; i < len(guardianList) && i < ELECTION_GUARDIAN_EXCELLENCE_MAX_NUMBER; i++ {
+		fmt.Printf("elections %10d rewards: top guardian %x, has %d votes\n", _getCurrentElectionBlockNumber(), guardianList[i].address, guardianList[i].vote)
 		totalVotes = safeuint64.Add(totalVotes, guardianList[i].vote)
-		topGuardiansStake[guardianList[i].guardian] = guardianList[i].vote
 	}
-	return
+	for i = ELECTION_GUARDIAN_EXCELLENCE_MAX_NUMBER; i < len(guardianList); i++ {
+		if guardianList[i].vote != guardianList[i-1].vote {
+			break
+		}
+		fmt.Printf("elections %10d rewards: top guardian %x, has %d votes\n", _getCurrentElectionBlockNumber(), guardianList[i].address, guardianList[i].vote)
+		totalVotes = safeuint64.Add(totalVotes, guardianList[i].vote)
+	}
+	if i < len(guardianList) {
```

Comment:
this if statement is completely not needed isn't it?
`i == len(guardianList)` ==> `guardianList[0:i] == guardianList`...

Reply:
if the i we stopped is less than the total length we return a slice
if not we return the whole slice so its needed